### PR TITLE
Feature/378/remove polyfills

### LINKED ISSUE #378: Can we remove the polyfills and es6.js files from alert banner?

### PR DESCRIPTION
**Pull request by markconroy**
Closes #378 

---
Thanks to [Big Blue Door](https://www.bigbluedoor.net/) for sponsoring my time to work on this.

### CODE REVIEW

File: js/alert_banner.js [hunk 1]
```diff
@@ -11,34 +11,34 @@
...
     document.cookie = 'hide-alert-banner-token=' + new_cookie + '; expires=' + new Date(expiry).toString() + '; SameSite=Lax;'
   }
 
   $(document).ready(function() {
 
-    var all_cookies = document.cookie.split('; ');
-    for (var i = 0; i < all_cookies.length; i++) {
-      var indv_cookie = all_cookies[i].split('=');
+    const all_cookies = document.cookie.split('; ');
+    for (const i = 0; i < all_cookies.length; i++) {
```
Comment (Adnan-cds) [inline js/alert_banner.js]: Hi Mark, this is raising an exception in the Javascript console as `i` here needs to be a variable rather than const :(

**Review by Adnan-cds — COMMENTED** [review]
#379 had dropped [modules/localgov_alert_banner_full_page/js/full-page-alert-banner.es6.js](https://github.com/localgovdrupal/localgov_alert_banner/pull/379/files#diff-13c3de9bf9feba9e13fe6685451371f4bfcc466f9df616b783580194a07dc3d2) although this one isn't.  I hope this is intentional.

File: js/alert_banner.js (hunk 1 above)
Comment (markconroy) [reply js/alert_banner.js]: That's updated now.

**Comment by Adnan-cds** [comment]
Andy, would you mind checking why a test is still [failing](https://github.com/localgovdrupal/localgov_alert_banner/actions/runs/11124795482/job/30911504225?pr=381#step:5:16).

> https://github.com/localgovdrupal/localgov_alert_banner/pull/379 had dropped [modules/localgov_alert_banner_full_page/js/full-page-alert-banner.es6.js](https://github.com/localgovdrupal/localgov_alert_banner/pull/379/files#diff-13c3de9bf9feba9e13fe6685451371f4bfcc466f9df616b783580194a07dc3d2) although this one isn't. I hope this is intentional.

Mark, would you please confirm this.

**Comment by andybroomfield** [comment]
@Adnan-cds I've updated this. With let / const, they are bound to the enclosing block, which is the for loop. Fix was to define it before the loop.

**Comment by Adnan-cds** [comment]
Thanks Andy.  I am just waiting for Mark to confirm about the [full-page-alert-banner.es6.js](https://github.com/localgovdrupal/localgov_alert_banner/pull/379/files#diff-13c3de9bf9feba9e13fe6685451371f4bfcc466f9df616b783580194a07dc3d2) file from the [previous pull request](https://github.com/localgovdrupal/localgov_alert_banner/pull/379).

**Comment by markconroy** [comment]
@Adnan-cds I don't understand the question? Is there something missing that I need to add, or something added that I need to remove? 

We don't need the .es6.js file any more, so can just transfer the code from it to the standard `full-page-alert-banner.js` file.

**Comment by Adnan-cds** [comment]
Hi Mark,
The previous [pull request](https://github.com/localgovdrupal/localgov_alert_banner/pull/379) removed the full-page-alert-banner.es6.js file.  But this pull request doesn't touch that file.  I just need a confirmation if this is intentional or not.

**Comment by markconroy** [comment]
I don't think I even noticed that, but it looks like there is no .es6.js file in the 1.8 branch so I guess it was removed already.

So I _think_ the answer is "Yes, this is intentional" 😄

**Comment by Adnan-cds** [comment]
Thanks.  All good then :)

**Review by Adnan-cds — APPROVED** [review]
Looks okay.

**Comment by andybroomfield** [comment]
As this is approved, shall I merge it so it can can into the 1.8 release?

**Comment by andybroomfield** [comment]
@Adnan-cds It looks like you already removed the es6 file
https://github.com/localgovdrupal/localgov_alert_banner/pull/356/commits/fb651c36f4c3e9bd50d03b19c68e2a32660f3805

**Pull request merged**